projects/aws-observability/aws-otel-collector/helm/patches/0002-namespace-patch.yaml

```diff
@@ -0,0 +1,193 @@
+index 0343681..e896140 100644
+--- a/charts/opentelemetry-collector/templates/podmonitor.yaml
++++ b/charts/opentelemetry-collector/templates/podmonitor.yaml
+@@ -3,6 +3,7 @@ apiVersion: monitoring.coreos.com/v1
+ kind: PodMonitor
+ metadata:
+   name: {{ include "opentelemetry-collector.fullname" . }}-agent
++  namespace: {{ .Release.Namespace | quote }}
+   labels:
+     {{- include "opentelemetry-collector.labels" . | nindent 4 }}
+     {{- range $key, $value := .Values.podMonitor.extraLabels }}
+diff --git a/charts/opentelemetry-collector/templates/prometheusrule.yaml b/charts/opentelemetry-collector/templates/prometheusrule.yaml
+index 2f2ffe7..e8de6e6 100644
+--- a/charts/opentelemetry-collector/templates/prometheusrule.yaml
++++ b/charts/opentelemetry-collector/templates/prometheusrule.yaml
+@@ -3,6 +3,7 @@ apiVersion: monitoring.coreos.com/v1
+ kind: PrometheusRule
+ metadata:
+   name: {{ include "opentelemetry-collector.fullname" . }}
++  namespace: {{ .Release.Namespace | quote }}
+   labels:
+     {{- include "opentelemetry-collector.labels" . | nindent 4 }}
+     {{- range $key, $value := .Values.prometheusRule.extraLabels }}
+diff --git a/charts/opentelemetry-collector/templates/service.yaml b/charts/opentelemetry-collector/templates/service.yaml
+index eeb1178..fc347a4 100644
+--- a/charts/opentelemetry-collector/templates/service.yaml
++++ b/charts/opentelemetry-collector/templates/service.yaml
+@@ -3,6 +3,7 @@ apiVersion: v1
+ kind: Service
+ metadata:
+   name: {{ include "opentelemetry-collector.fullname" . }}
++  namespace: {{ .Release.Namespace | quote }}
+   labels:
+     {{- include "opentelemetry-collector.labels" . | nindent 4 }}
+     {{- include "opentelemetry-collector.component" . | nindent 4 }}
+diff --git a/charts/opentelemetry-collector/templates/serviceaccount.yaml b/charts/opentelemetry-collector/templates/serviceaccount.yaml
+index 5daa658..00cd049 100644
+--- a/charts/opentelemetry-collector/templates/serviceaccount.yaml
++++ b/charts/opentelemetry-collector/templates/serviceaccount.yaml
+@@ -3,6 +3,7 @@ apiVersion: v1
+ kind: ServiceAccount
+ metadata:
+   name: {{ include "opentelemetry-collector.serviceAccountName" . }}
++  namespace: {{ .Release.Namespace | quote }}
+   labels:
+     {{- include "opentelemetry-collector.labels" . | nindent 4 }}
+   {{- with .Values.serviceAccount.annotations }}
+diff --git a/charts/opentelemetry-collector/templates/servicemonitor.yaml b/charts/opentelemetry-collector/templates/servicemonitor.yaml
+index dae68d2..005efda 100644
+--- a/charts/opentelemetry-collector/templates/servicemonitor.yaml
++++ b/charts/opentelemetry-collector/templates/servicemonitor.yaml
+@@ -3,6 +3,7 @@ apiVersion: monitoring.coreos.com/v1
+ kind: ServiceMonitor
+ metadata:
+   name: {{ include "opentelemetry-collector.fullname" . }}
++  namespace: {{ .Release.Namespace | quote }}
```

Comment:
We probably want a default namespace for all those namespaced resources.

Reply:
Added. I'm naming it `observability` for now, but we can change it later if we find a name for better fit.